packages/wonder-blocks-core/src/hooks/__tests__/use-latest-ref.test.ts

```diff
@@ -37,4 +37,12 @@ describe("useLatestRef", () => {
         // Assert
         expect(result.current).toBe(refFromFirstRender);
     });
+
+    // Type tests
+    const _ = () => {
+        // The correct, non-nullable type should be inferred for `current`.
+        useLatestRef(123) satisfies {current: number};
+        // @ts-expect-error the result should not be assignable to the wrong type
+        useLatestRef(123) satisfies {current: string};
+    };
```

Comment:
Is the convention to put type tests in a `*.typestest.ts` file? I see a couple such files. @somewhatabstract

Reply:
Yes, that's the convention we've used for types-only testing.